src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/CSharpCodeParser.cs

```diff
@@ -2472,7 +2476,7 @@ private void ParseUsingDeclaration(in SyntaxListBuilder<RazorSyntaxNode> builder
             var nonNamespaceTokenCount = TokenBuilder.Count;
             AcceptWhile(IsSpacingTokenIncludingComments);
             var start = CurrentStart;
-            if (At(SyntaxKind.Identifier))
+            if (At(SyntaxKind.Identifier) || At(CSharpSyntaxKind.GlobalKeyword))
```

Comment:
Because `using global::<type-name>` is a very common pattern; most projects will have some with that syntax automatically generated and then included in every file. Previously, `global` was treated as an identifier by the lexer, whereas it's now correctly treated as a keyword.